leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -236,43 +236,63 @@ public TaggedLogger(String name, String tag) {
         }
 
         public void trace(String s, Throwable t) {
-            delegate.trace(tag + s, t);
+            if (delegate.isTraceEnabled()) {
+                delegate.trace(tag + s, t);
+            }
         }
 
         public void debug(String s, Throwable t) {
-            delegate.debug(tag + s, t);
+            if (delegate.isDebugEnabled()) {
+                delegate.debug(tag + s, t);
+            }
         }
 
         public void info(String s, Throwable t) {
-            delegate.info(tag + s, t);
+            if (delegate.isInfoEnabled()) {
+                delegate.info(tag + s, t);
+            }
         }
 
         public void warn(String s, Throwable t) {
-            delegate.warn(tag + s, t);
+            if (delegate.isWarnEnabled()) {
+                delegate.warn(tag + s, t);
+            }
         }
 
         public void error(String s, Throwable t) {
-            delegate.error(tag + s, t);
+            if (delegate.isErrorEnabled()) {
```

Comment:
Probably not. I _guess_ the underlying logger could be set to `FATAL` but (a) it's unlikely, and (b) the ERROR logging isn't going to kill perf. I guess I just did this for consistency.